### ISSUE TITLE
loaded component names not updated, if there are no substantive package level changes ...

### ISSUE THREAD

**Issue by dalehenrich** [GemTalk/Rowan#630]
The test  RwGsTopazRowanToolTest>>testTopazAddToplevelComponent illustrates the problem (x is expected to equal y):
```
	|  project projectName componentName1 componentName2 packageName1 packageName2
		x y |
	projectName := 'TopazApiProject'.
	componentName1 := 'Core'.
	componentName2 := 'Test'.
	packageName1 := projectName , '-Core'.
	packageName2 := projectName, '-Tests'.

"Create project and add new component"
	project := self _createTestProjectNamed: projectName.

	Rowan gemstoneTools topaz
		addTopLevelComponentNamed: componentName2
			forProjectNamed: projectName 
			condition: 'test';
		addNewPackageNamed: packageName2 
			forProjectNamed: projectName 
			toComponentNamed: componentName2;
		yourself.

"Validate"
	self assert: (x := project componentNames sort)  = (y := {componentName1 . componentName2 } sort).
	self assert: (project componentForPackageNamed: packageName1) name = componentName1.
	self assert: (project componentForPackageNamed: packageName2) name = componentName2.
```

**Comment by dalehenrich** [comment]
Worth noting that all of the packages in this test are empty ... so there is not much going on of substance

**Comment by dalehenrich** [comment]
fixed the problem as part of c78ecb806 ... turns out that on a load the _loadSpecification used for the load, was not being recorded with the loaded package ...
